gui/src/containers/Bridges/Edit.js

```diff
@@ -24,19 +24,18 @@ const SuccessNotification = ({name}) => (<React.Fragment>
   Successfully updated <Link to={`/bridges/${name}`}>{name}</Link>
 </React.Fragment>)
 
-export const Edit = useHooks((props) => {
+export const Edit = useHooks(props => {
+  useEffect(() => {
+    const {fetchBridgeSpec, match} = props
+    fetchBridgeSpec(match.params.bridgeId)
+  }, [])
   const {bridge, updateBridge} = props
-  const checkLoaded = () => props.bridge
+  const checkLoaded = () => bridge
   const onLoad = (buildLoadedComponent) => {
     if (checkLoaded()) return buildLoadedComponent(props)
```

Comment:
I know this existed prior to your refactor but I would like to walk through it to make sure it's clear.

1. `const checkLoaded = () => bridge` is defined at the component creation level.
2. An `Edit` instance get created when `props` is changed
3. `checkLoaded()` gets invoked to ultimately return what is `props.bridge`

My question is, can't we just check `bridge`? Why must we invoke the method `checkLoaded`? Is that because `mapStateToProps` can update the `bridge` underneath the instance? But wouldn't that trigger a rerender? Furthermore, this is only invoked `onLoad`.

Reply:
@rupurt can correct me if I'm wrong but I think it's there only for readability. removing `checkLoaded` and switching conditional to `if(bridge)` from `if(checkLoaded())` shouldn't change anything

Reply:
It's there for readability.